components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -22,8 +23,8 @@ import java.util.WeakHashMap
  */
 class ObserverRegistry<T> : Observable<T> {
     private val observers = mutableSetOf<T>()
-    private val lifecycleObservers = WeakHashMap<T, LifecycleBoundObserver<T>>()
-    private val viewObservers = WeakHashMap<T, ViewBoundObserver<T>>()
+    private val lifecycleObservers = WeakHashMap<T, WeakReference<LifecycleBoundObserver<T>>>()
+    private val viewObservers = WeakHashMap<T, WeakReference<ViewBoundObserver<T>>>()
```

Comment:
Do we really need a `WeakReference` here?

Wouldn't it be enough (and more explicit) to remove the entries in `unregister()`?

Reply:
And the explicit "remove" may be easier to test too :)